Rename path type to recursive relationship (#217)
kuzudb/kuzu-docs: src/content/docs/cypher/data-types/index.mdx

----------------------------------------------------
```

> ## RELATIONSHIP

Reviewer: ```suggestion
## REL
```

| Size | Description
| --- | ---
| fixed | represents a relationship in a graph

> `RELATIONSHIP` is a logical type. Internally, `RELATIONSHIP` is processed as `STRUCT` type. A `RELATIONSHIP` always contains a

Reviewer: ```suggestion
`REL` is a logical type that represents a relationship (i.e., an edge). Internally, `REL` is processed as `STRUCT` type. A `REL` always contains a
```

src ID field with key `_SRC`, a dst ID field with key `_DST`, an internal ID field with key `_ID`
and a label field with key `_LABEL`. The rest fields are rel properties.

> Here's how to return `RELATIONSHIP` column:

Reviewer: ```suggestion
Here's how to return a relationship column that's of type `REL`:
```

----------------------------------------------------------------------------------
```

> ## RECURSIVE RELATIONSHIP

Reviewer: ```suggestion
## RECURSIVE_REL
```

> `RECURSIVE RELATIONSHIP` is a logical type. Internally, `RECURSIVE RELATIONSHIP` is processed as `STRUCT` type, more specifically, a

Reviewer: ```suggestion
`RECURSIVE_REL` is a logical type that represents recursive relationships. i.e., paths of arbitrary lengths. Internally, `RECURSIVE_REL` is processed as `STRUCT` type, more specifically, a
```

> `STRUCT{LIST[NODE], LIST[RELATIONSHIP]}`. A `RECURSIVE RELATIONSHIP` always contains a nodes field with the key `_NODES` and a

Reviewer: ```suggestion
`STRUCT{LIST[NODE], LIST[REL]}`. A `RECURSIVE_REL` always contains a nodes field with the key `_NODES` and a
```

relationships field with the key `_RELS`.

> #### Return `RECURSIVE RELATIONSHIP` column

Reviewer: ```suggestion
#### Return a column that's of type `RECURSIVE_REL`
```
